java/client/src/main/java/glide/connectors/handlers/ChannelHandler.java

```diff
@@ -82,4 +87,18 @@ public ChannelFuture close() {
         callbackDispatcher.shutdownGracefully();
         return channel.close();
     }
+
+    /** Propagate an error from Netty's future into the future returned to user. */
+    @RequiredArgsConstructor
+    private static class NettyFutureErrorHandler implements ChannelFutureListener {
+
+        private final CompletableFuture<Response> promise;
+
+        @Override
+        public void operationComplete(@NonNull ChannelFuture channelFuture) throws Exception {
+            if (!channelFuture.isSuccess()) {
```

Comment:
According to https://netty.io/4.0/api/io/netty/channel/ChannelFuture.html, !isSuccess just means that the future did not complete successfully.  
We should check for `cause != null` to determine if it failed.  
In the case of `isCancelled`, do we feel it is necessary to terminate the client?

Reply:
I added extra checks, but I have no idea how to test them. Any clue?